src/openrct2/scripting/ScContext.hpp

```diff
@@ -0,0 +1,302 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#ifdef __ENABLE_SCRIPTING__
+
+#    include "../actions/CustomAction.hpp"
+#    include "../actions/ParkSetNameAction.hpp"
+#    include "../actions/SmallSceneryPlaceAction.hpp"
+#    include "../object/ObjectManager.h"
+#    include "../scenario/Scenario.h"
+#    include "Duktape.hpp"
+#    include "HookEngine.h"
+#    include "ScConfiguration.hpp"
+#    include "ScDisposable.hpp"
+#    include "ScObject.hpp"
+#    include "ScriptEngine.h"
+
+#    include <cstdio>
+#    include <memory>
+
+namespace OpenRCT2::Scripting
+{
+    class ScContext
+    {
+    private:
+        ScriptExecutionInfo& _execInfo;
+        HookEngine& _hookEngine;
+
+    public:
+        ScContext(ScriptExecutionInfo& execInfo, HookEngine& hookEngine)
+            : _execInfo(execInfo)
+            , _hookEngine(hookEngine)
+        {
+        }
+
+    private:
+        std::shared_ptr<ScConfiguration> configuration_get()
+        {
+            return std::make_shared<ScConfiguration>();
+        }
```

Comment:
What purpose is this serving?

Reply:
It is an API binding for getting the configuration object.

Reply:
ok